site/cypress/e2e/Cypress-Gradeable/rainbow_grading.spec.js

```diff
@@ -0,0 +1,71 @@
+
+    });
+    it('Enable viewing of rainbow grades and generating the rainbow grading', () => {
+        cy.get('[data-testid="display-rainbow-grades-summary"]').check();
+        cy.get('[data-testid="display-rainbow-grades-summary"]').should('be.checked');
+        cy.visit(['sample', 'reports', 'rainbow_grades_customization']);
+        cy.get('[data-testid="display-grade-summary"]').check();
+        cy.get('[data-testid="display-grade-summary"]').should('be.checked');
+        cy.get('#display_grade_details').check();
+        cy.get('[data-testid="display-benchmarks-average"]').check();
+        cy.get('[data-testid="display-benchmarks-stddev"]').check();
+        cy.get('[data-testid="display-benchmarks-perfect"]').check();
+        cy.get('[data-testid="save-status-button"]').click();
+        cy.get('[data-testid="save-status"]', { timeout: 15000 }).should('contain', 'Rainbow grades successfully generated!');
+        cy.visit(['sample', 'grades']);
+        ['USERNAME', 'NUMERIC ID', 'AVERAGE', 'STDDEV', 'PERFECT'].forEach((fields) => {
+            cy.get('[data-testid="rainbow-grades"]').should('contain', fields);
+        });
+        cy.get('[data-testid="rainbow-grades"]').should('contain', 'Information last updated');
+        ['ta', 'student', 'grader', 'instructor'].forEach((username) => {
+            cy.logout();
+            cy.login(username);
+            cy.visit(['sample', 'grades']);
+            cy.get('[data-testid="rainbow-grades"]').should('contain', `Lecture Participation Polls for: ${username}`);
+            if (username==='instructor') {
+                checkRainbowGrades( 'instructor', 801516157, 'Quinn', 'Instructor');
+                checkRainbowGradesOption();
+            }
+            else if (username === 'ta') {
+                checkRainbowGrades('ta', 281179137, 'Jill', 'TA');
+                checkRainbowGradesOption();
+            }
+            else if (username === 'student') {
+                checkRainbowGrades('student', 'student', 410853871, 'Joe', 'Student');
+                checkRainbowGradesOption();
+            }
+            else if (username === 'grader') {
+                checkRainbowGrades('grader', 10306042, 'Tim', 'Grader');
+                checkRainbowGradesOption();
+            }
+        });
+        cy.visit(['sample', 'config']);
+        cy.get('[data-testid="display-rainbow-grades-summary"]').uncheck();
+        cy.get('[data-testid="display-rainbow-grades-summary"]').should('not.be.checked');
+        cy.visit(['sample', 'grades']);
+        cy.get('[data-testid="rainbow-grades"]').should('contain', 'No grades are available...');
+    });
+});
+
+const checkRainbowGrades = (username, numericId, firstName, lastname) => {
+    [username, numericId, firstName, lastname].forEach((value) => {
+        cy.get('[data-testid="rainbow-grades"]').should('contain', value);
+    });
+};
+const checkRainbowGradesOption = () => {
+    cy.get('[data-testid="rainbow-grades"]')
```

Comment:
Would it be more consistent with this file to use a .forEach() statement on an array instead of a bunch of .and() statements? Not that it is bad practice, just wondering for consistency.